go-controller/pkg/cni/cniserver.go

```diff
@@ -93,6 +94,15 @@ func NewCNIServer(rundir string, useOVSExternalIDs bool, factory factory.NodeWat
 	}).Methods("POST")
 
 	factory.AddPodHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			newPod := newObj.(*kapi.Pod)
+			newAnnot, newErr := util.UnmarshalPodAnnotation(newPod.Annotations)
+			oldPod := oldObj.(*kapi.Pod)
+			oldAnnot, oldErr := util.UnmarshalPodAnnotation(oldPod.Annotations)
+			if newErr == nil && oldErr == nil && oldAnnot.MAC.String() != newAnnot.MAC.String() {
```

Comment:
you could also just check that len(oldAnnot.MAC.String()) >0  to avoid trying to cancel an PR where the mac is being added to a pod right?

Reply:
@trozet not sure I understand the comment... UnmarshalPodAnnotation() will fail with an error if the MAC address isn't valid:

```
	podAnnotation.MAC, err = net.ParseMAC(a.MAC)
	if err != nil {
		return nil, fmt.Errorf("failed to parse pod MAC %q: %v", a.MAC, err)
	}
```

So we can't check len(oldAnnot.MAC.String()) since that would dereference NULL in the error case. So we have to check `err` anyway, and if `err == nil` we know that the MAC is valid and len() > 0